### ISSUE TITLE
Sync status is always "syncing" when node is alone or low blocks count

### ISSUE THREAD

**Issue by cupuyc** [ethereum/ethereumj#605]
When testing private networks I found 2 cases when in ethereumJ core `syncManager.isSyncDone()` is always false:

1. When I run single node without peers with no DB

1. When I run node with 1 peer and have synced all 220 blocks

In described cases geth returns `eth.syncing=false` properly. While ethereumJ returns 
```
{
  "startingBlock": "0x0",
  "currentBlock": "0x0",
  "highestBlock": "0x0"
}
```

I used `develop` version. And noticed similar behavior in `master`.

**Comment by Nashatyrev** [comment]
> When I run single node without peers with no DB
Not sure what `syncManager.isSyncDone()` should return in this case... 
Will try to reproduce the second case on the latest `develop`

**Comment by Nashatyrev** [comment]
Couldn't reproduce the second case with `PrivateMinerSample`: 
- Run sample, wait for several blocks to be mined 
- Stop and remove `sampleDB-2` dir (the regular node db)
- Start again, the regular node is synced up and starts issuing txs (triggered on `onSyncDone`)
- Tried with different chain lengths

**Comment by cupuyc** [comment]
I checked with your scenario.

We discussed on meeting drawback if current sync done logic. Sync done event will only happen  when all blocks downloaded and next block received from network. According that spec all works fine.

@Nashatyrev 
1. Both my issues depend on case when there is no active miner present. I propose to take no actions for my issues as this is non standard scenario.

2. I propose not to use `sync.enabled=false` on mining nodes. This might confuse end users, which copy past samples.

If there are few miners - newly added miner should sync current state first and only then start mining.
